src/channel/coolholegolds.js

```diff
@@ -0,0 +1,76 @@
+var ChannelModule = require("./module");
+const LOGGER = require("@calzoneman/jsli")("channel");
+
+/**
+ * A coolhole module.
+ * This module calculates if chat messages are gold.
+ * @param {Object} _channel 
+ */
+function CoolholeGoldsModule(_channel) {
+    ChannelModule.apply(this, arguments);
+    this.supportsDirtyCheck = false;
+}
+
+CoolholeGoldsModule.prototype = Object.create(ChannelModule.prototype);
+
+/**
+ * Calculates if the chat message is gold, and applies it to data.meta.coolholeMeta
+ * @param {Object} msgobj message object about to be sent out to clients
+ * @returns always return true to let the chat message passthrough
+ */
+CoolholeGoldsModule.prototype.calculateGold = function (msgobj) {
+    if(this.channel.modules["playlist"].current === null ||
+       this.channel.modules["playlist"].current.media === null ||
+       this.channel.modules["playlist"].current.media.title === null)
+       return;
+
+    // Get current video title
+    const currentVideoTitle = this.channel.modules["playlist"].current.media.title;
+    
+    // Add some date specific modifer to let golds "go stale"; this is based on each quarter per year (~3 month period)
+    const dateModifier = Math.floor((new Date().getUTCMonth() + 3) / 3) + new Date().getUTCFullYear();
+
+    // Convert to lower case to allow for different cased gold
+    const chatText = msgobj.msg.toLowerCase();
+
+    // Combine the msg, video title, date, and a constant (legacy was "co"); 
+    const lotteryText = chatText + currentVideoTitle + "co" + dateModifier.toString();
+
+    // Hash string
+    let lotteryHash = hashFunc(lotteryText);
+
+    // Modulo hash by 100
+    lotteryHash %= 100;
+
+    // No negatives
+    lotteryHash = lotteryHash < 0 ? -lotteryHash : lotteryHash;
+
+    // 1% chance? Idk either. If it's equal to one, they get a gold
+    if (lotteryHash === 1) {
+        LOGGER.info(`calculateGold: Found Gold Message: ${chatText}`);
+        msgobj.meta.coolholeMeta.otherClasses.push("text-lottery");
```

Comment:
question: do we wanna add european gold here too or leave that in js/css for now?

Reply:
Huh, I didn't think about that actually. Maybe it would be cleaner on the server side?

We can probably just leave it the way it is for now...there are other features like points, sounds, options that should probably be completed before we refactor a bit.

Reply:
Yeah I agree. Let's work towards an MVP and get something close to how CH works currently (custom JS and all). Then we can start porting over mainstay features into source

Reply:
Yeah I agree. Let's work towards an MVP and get something close to how CH works currently (custom JS and all). Then we can start porting over mainstay features into source